### PR TITLE
Search function

### LINKED ISSUE #64: Search function
## User story

As a **user**, I want to be able to search my entries to get to important data faster and see only what is relevant for me

## ToDos

- [x] Add search component
- [x] Add functionality to search component
- [x] Filter results in medics, doctors and diaries
- [x] Add search function to medics, doctors and diaries

### PR DESCRIPTION
**Pull request by slawoe**
Implement a search function for doctors, medics and diary entries

Fix #64

### CODE REVIEW

File: client/src/pages/DiaryList.js [hunk 1]
```diff
@@ -19,19 +20,27 @@ function DiaryList() {
     showDiaryEntries();
   }, []);
 
+  const filteredDiary = diaryEntries?.filter((entries) => {
+    return entries.date.startsWith(query);
+  });
+
```
Comment (manuelpapa) [inline client/src/pages/DiaryList.js:26]: I'm happy for you that you finally made a very smart and short code for your filtering

File: client/src/pages/DiaryList.js [hunk 2]
```diff
@@ -19,19 +20,27 @@ function DiaryList() {
...
+            placeholder="Tagebucheintrag suchen"
+            value={query}
+            onChange={(value) => setQuery(value)}
+          />
         </>
       }
       childrenmainsection={
         <>
-          {diaryEntries?.map((diaryentry) => (
+          {filteredDiary?.map((diaryentry) => (
```
Comment (manuelpapa) [inline client/src/pages/DiaryList.js:43]: maybe `diaryEntry` here?

File: client/src/pages/doctors/DoctorsList.js [hunk 3]
```diff
@@ -22,12 +23,22 @@ function Doctors() {
...
+      childrenheadsection={
+        <Searchbar
+          placeholder={"Arzt suchen"}
+          value={query}
+          onChange={(value) => setQuery(value)}
+        />
+      }
       childrenmainsection={
         <ContentWithAddFunction
           addcomponent={
```
Comment (manuelpapa) [inline client/src/pages/doctors/DoctorsList.js:44]: Check here if you're able to find a consistent way in the naming (`Pascal`, `CamelCase`?).

**Review by manuelpapa — COMMENTED** [review]
Slawo, I don't know, what to add. Your code looks so clever and you even managed to shorten some passages, Very pleasant!

File: client/src/pages/DiaryList.js (hunk 1 above)
Comment (slawoe) [reply client/src/pages/DiaryList.js:26]: mille grazie!

File: client/src/pages/DiaryList.js (hunk 2 above)
Comment (slawoe) [reply client/src/pages/DiaryList.js:43]: great -> implemented

File: client/src/pages/doctors/DoctorsList.js (hunk 3 above)
Comment (slawoe) [reply client/src/pages/doctors/DoctorsList.js:44]: You're right - will fix this while refactoring in the next days

**Pull request merged**